pkg/cache/cache.go

```diff
@@ -131,15 +131,14 @@ func (c *ClusterQueue) update(in *kueue.ClusterQueue) error {
 			continue
 		}
 
-		usedFlavors, exist := c.UsedResources[r.Name]
-		if !exist {
-			usedFlavors = make(map[string]int64, len(r.Flavors))
+		existingUsedFlavors := map[string]int64{}
+		if existing, ok := c.UsedResources[r.Name]; ok {
+			existingUsedFlavors = existing
```

Comment:
why not just?

```golang
existingUsedFlavors := c.UsedResources[r.Name]
```

Reply:
wouldn't that return a nil if not exist?

Reply:
yes, and that's fine https://go.dev/play/p/ArI2F_FzrhA 
Sometimes golang feels like python :upside_down_face:

Reply:
sg, updated and squashed.